src/data_source/storage/sql.rs

```diff
@@ -469,21 +467,29 @@ impl PrunedHeightStorage for SqlStorage {
     async fn save_pruned_height(&mut self, height: u64) -> Result<(), Self::Error> {
         // id is set to 1 so that there is only one row in the table.
         // height is updated if the row already exists.
-        self.query_opt(
-            "INSERT INTO pruned_height (id, last_height) VALUES (1, $1) ON CONFLICT (id) DO UPDATE SET last_height = $1;",
-            &[&(height as i64)],
-        )
-        .await?;
-        Ok(())
+        self.transaction()
+            .await?
+            .upsert(
+                "pruned_height",
+                ["id", "last_height"],
+                ["id"],
+                [[sql_param(&(1_i32)), sql_param(&(height as i64))]],
+            )
+            .await?;
+
+        self.commit().await.map_err(|err| QueryError::Error {
```

Comment:
Commit should probably be called at the higher level, as we do elsewhere. E.g. in the pruner task it could be like
* prune
* save pruned height
* commit